### DISCUSSION TITLE
Difference between rootOrgId and orgId

### DISCUSSION THREAD

**Discussion by githubpiyush** [Sunbird-coKreat/Community#34] [General Discussions]
We are getting two different attributes for the organisation ID from the following APIs in the staging environment:

1. /api/user/v3/search  This API has a rootOrgId attribute for the organisation ID.
2. /api/reg/search This API has an orgId attribute for the organisation ID.

What is the significance of these attributes ?
How are these attributes related to each other?

cc - @kiranharidas187 @vijiurs @aks30 @pradeepkumarcm96

**Comment by pallakartheekreddy** [comment]
@githubpiyush In coKreat workflows, we have two types of organization IDs required. Both organization ID for different purposes one is sourcing organization ID and the other one is contribution organization ID.
1. `/api/user/v3/search` API has a `rootOrgId` attribute this we are using as sourcing organization ID.
2. `/api/reg/search` API has an orgId attribute this we are using as a contribution organization ID.

There is no relation between these two attributes. Please [click here ](https://project-sunbird.atlassian.net/wiki/spaces/SingleSource/pages/1960574996/User+Roles) to know more details about each user in coKreat.

**Reply by githubpiyush** [reply]
@pallakartheekreddy @Nadiya496 

Suppose a user is invited by the contributor org admin (this org is present in consumption).
Now the external user register on the VDN portal his information will be stored in the open saber registry API, this user will have a orgId in the open saber registry API. 

So in this case we will have some relation between rootOrgId and orgId or not?

**Reply by pallakartheekreddy** [reply]
@githubpiyush When contributor org admin entry is created using open saber registry API for the existing rootOrgId we store that rootOrgId in the `Org` table and the column name is `orgId`. This column will be empty for other org admins who are choosing as orgadmin in the VDN portal for the first time.
